plugins/woocommerce-admin/client/marketplace/components/category-selector/category-selector.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Spinner } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import CategoryLink from './category-link';
+import './category-selector.scss';
+import CategoryDropdown from './category-dropdown';
+
+export type Category = {
+	readonly slug: string;
+	readonly label: string;
+	selected: boolean;
+};
+
+export type CategoryAPIItem = {
+	readonly slug: string;
+	readonly label: string;
+};
+
+function fetchCategories(): Promise< CategoryAPIItem[] > {
+	return fetch(
+		'https://woocommerce.com/wp-json/wccom-extensions/1.0/categories'
```

Comment:
There are about 6 instances of `https://woocommerce.com` addresses in the marketplace folder. I think we should move them into `marketplace/components/constants.ts`, to make it easier to change the URLs when the domain changes, and to update the endpoint addresses when needed.

Reply:
OK, done in https://github.com/woocommerce/woocommerce/pull/39561/commits/62289f4eed5114d245a7dd1a6209518e0d7dd700